src/_pytest/pathlib.py

```diff
@@ -49,10 +49,7 @@ def on_rm_rf_error(func, path: str, exc, *, start_path):
         return
 
     if func not in (os.rmdir, os.remove, os.unlink):
-        warnings.warn(
-            PytestWarning("(rm_rf) error removing {}: {}".format(path, excvalue))
-        )
-        return
+        raise
```

Comment:
```
        if func not in (os.rmdir, os.remove, os.unlink):
>           raise
E           RuntimeError: No active exception to reraise
```

Did you mean `raise RuntimeError("unexpected function passed to `rm_rf`: {}".format(func))` or something like this?

Also I'm not sure this would be a good approach, because I've seen this warning on occasion, this change would change it into a hard error instead.

Also just noticed that one of the errors might be `FileNotFoundError` on my system:

---

```
...\_pytest\pathlib.py:47: PytestWarning: (rm_rf) error removing c:\...\pytest-of-Bruno\garbage-f0c355a1-181a-4771-a379-6f7d7d82390a\popen-gw3\test_pytest_plugins_in_non_top_level_conftest_unsupported_no_top_level_conftest0\subdirectory\__pycache__\conftest.cpython-36-pytest-5.2.2.dev43+g2e11ea610.d20191019.pyc: [WinError 3] O sistema não pode encontrar o caminho especificado: '<path repeats>'
```

"O sistema não pode encontrar o caminho especificado" means "System could not find the specified path", which makes sense if `rm_rf` is being called by multiple processes at the same time.